sdk/storage/azure-storage-blob-stress/src/main/java/com/azure/storage/blob/stress/DownloadToFile.java

```diff
@@ -0,0 +1,151 @@
+        this.blobPrintableSize = (int) Math.min(options.getSize(), 1024);
+        this.asyncNoFaultClient = getAsyncContainerClientNoFault().getBlobAsyncClient(options.getBlobName());
+        this.syncClient = getSyncContainerClient().getBlobClient(options.getBlobName());
+        this.asyncClient = getAsyncContainerClient().getBlobAsyncClient(options.getBlobName());
+    }
+
+    @Override
+    protected boolean runInternal(Context span) {
+        Path downloadPath = directoryPath.resolve(UUID.randomUUID() + ".txt");
+        BlobDownloadToFileOptions blobOptions = new BlobDownloadToFileOptions(downloadPath.toString());
+
+        try {
+            syncClient.downloadToFileWithResponse(blobOptions, Duration.ofSeconds(options.getDuration()), span);
+            return validateDownloadedContents(downloadPath);
+        } finally {
+            deleteFile(downloadPath);
+        }
+    }
+
+    @Override
+    protected Mono<Boolean> runInternalAsync(Context span) {
+        Path downloadPath = directoryPath.resolve(UUID.randomUUID() + ".txt");
+        BlobDownloadToFileOptions blobOptions = new BlobDownloadToFileOptions(downloadPath.toString());
+
+        return asyncClient
+                .downloadToFileWithResponse(blobOptions)
+                .flatMap(ignored -> validateDownloadedContentsAsync(downloadPath, span))
+                .doFinally(i -> deleteFile(downloadPath));
+    }
+
+    private static void deleteFile(Path path) {
+        try {
+            path.toFile().delete();
+        } catch (Exception e) {
+            LOGGER.atInfo()
+                .addKeyValue("path", path)
+                .log("failed to delete file", e);
+        }
+    }
+
+    private boolean validateDownloadedContents(Path downloadPath) {
+        // Use crc to check for file mismatch, avoiding every parallel test streaming original data from disk
+        // If there's a mismatch, the original data can be streamed to check where the fault occurred
+        // Data is streamed in the first place to avoid holding potentially gigs in memory
+        long length = 0;
+        ByteBuffer contentHead = ByteBuffer.allocate(blobPrintableSize);
+        CRC32 dataCrc = new CRC32();
+        try (InputStream file = Files.newInputStream(downloadPath)) {
+            byte[] buf = new byte[4 * 1024 * 1024];
+            int read;
+            while ((read = file.read(buf)) != -1) {
+                dataCrc.update(buf, 0, read);
+                if (contentHead.hasRemaining()) {
+                    contentHead.put(buf, 0, Math.min(read, contentHead.remaining()));
+                }
+                length += read;
+            }
+        }
+        catch (IOException e) {
+            throw LOGGER.logExceptionAsError(new UncheckedIOException(e));
+        }
+
+        return ORIGINAL_CONTENT.checkMatch(dataCrc, length, contentHead);
+    }
+
+    private Mono<Boolean> validateDownloadedContentsAsync(Path downloadPath, Context span) {
```

Comment:
Do we need this method? The logic is the same as the synchronous version and in the async test we can just change the `flatMap` to `map` and use the synchronous version. From what I can tell I don't think this helps preventing blocks on the runner thread and just ends up being duplicate logic.

Reply:
kept this one (moved to `OriginalContent` though), but got rid of sync version of it in favor of `block()`